server/src/test/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesActionTests.java

```diff
@@ -220,7 +225,91 @@ public void handleException(TransportException exp) {
         }
     }
 
-    public void testGetsDiscoveredNodes() throws InterruptedException {
+    public void testGetsDiscoveredNodesWithZeroTimeout() throws InterruptedException {
+        setupGetDiscoveredNodesAction();
+        final GetDiscoveredNodesRequest getDiscoveredNodesRequest = new GetDiscoveredNodesRequest();
+        getDiscoveredNodesRequest.setWaitForNodes(2);
+        getDiscoveredNodesRequest.setTimeout(TimeValue.ZERO);
+        assertWaitConditionMet(getDiscoveredNodesRequest);
+    }
+
+    public void testGetsDiscoveredNodesByAddress() throws InterruptedException {
+        setupGetDiscoveredNodesAction();
+        final GetDiscoveredNodesRequest getDiscoveredNodesRequest = new GetDiscoveredNodesRequest();
+        getDiscoveredNodesRequest.setRequiredNodes(
+            Arrays.asList(localNode.getAddress().toString(), otherNode.getAddress().toString())
+        );
+        getDiscoveredNodesRequest.setTimeout(TimeValue.ZERO);
+        assertWaitConditionMet(getDiscoveredNodesRequest);
+    }
+
+    public void testGetsDiscoveredNodesByName() throws InterruptedException {
+        setupGetDiscoveredNodesAction();
+        final GetDiscoveredNodesRequest getDiscoveredNodesRequest = new GetDiscoveredNodesRequest();
+        getDiscoveredNodesRequest.setRequiredNodes(
+            Arrays.asList(localNode.getName(), otherNode.getName())
+        );
+        getDiscoveredNodesRequest.setTimeout(TimeValue.ZERO);
+        assertWaitConditionMet(getDiscoveredNodesRequest);
+    }
+
+    public void testGetsDiscoveredNodesByIP() throws InterruptedException {
```

Comment:
This test seems incorrect. The requirement here is `["0.0.0.0", "0.0.0.0"]` so both nodes match both addresses so this should fail.

Reply:
Right because we put these in a `Set` so the duplication goes away ... maybe we want to just fail a duplicate IP setting right away?

Reply:
Otherwise a duplicate IP in the setting means blacklisting that IP ... that's weird? :D

Reply:
Yes, I think that's right. Putting the same entry in twice is a misconfiguration, and bootstrapping is a dangerous time for a cluster, so I do not think we should be lenient.